src/dos/drive_fat.cpp

```diff
@@ -1306,27 +1306,31 @@ bool fatDrive::FindNextInternal(uint32_t dirClustNumber, DOS_DTA& dta,
 	trimString(&find_name[0], sizeof(find_name));
 	trimString(&extension[0], sizeof(extension));
 
-	// if(!(sectbuf[entryoffset].attrib & FatAttributeFlags::Directory))
-	if (extension[0]!=0) {
+	const auto entry_attributes = FatAttributeFlags(sectbuf[entryoffset].attrib);
+
+	// if(!entry_attributes.directory)
+
+	if (extension[0] != 0) {
 		safe_strcat(find_name, ".");
 		safe_strcat(find_name, extension);
 	}
 
-	/* Compare attributes to search attributes */
-	FatAttributeFlags attr_mask = {};
-	attr_mask.directory = true;
-	attr_mask.volume    = true;
-	attr_mask.system    = true;
-	attr_mask.hidden    = true;
-
 	// TODO What about volume/directory attributes?
 	if (attrs == FatAttributeFlags::Volume) {
-		if (!(sectbuf[entryoffset].attrib & FatAttributeFlags::Volume)) {
+		if (!entry_attributes.volume) {
 			goto nextfile;
 		}
 		dirCache.SetLabel(find_name, false, true);
 	} else {
-		if (~(attrs._data) & sectbuf[entryoffset].attrib & attr_mask._data) {
+		// Compare attributes to search attributes
+		// Compare attributes to search attributes
```

Comment:
(repeating comment; Sorry for this - probably a typo on my side)

Reply:
OK, fixed.